### PR TITLE
CI task for testing the down migrations of our database migrations

### PR DESCRIPTION
**Pull request by tirthyme**
#### Description:
- Adds a CI task and a shell script to verify if newly added migrations are working as expected and don't lead to dirty migrations.

### CODE REVIEW

**Comment by drpaneas** [comment]
Looks pretty straight forward. I had only one comment, but apart from that looks ready.

File: utilities/db-migration/migrate/migrate_db.go [hunk 1]
```diff
@@ -46,6 +46,16 @@ func Migrate(opType string, migrationPath string) error {
...
 		return nil
 
+	} else if opType == "downgrade_migration" {
+		if err := m.Steps(-1); err != nil {
+			return fmt.Errorf("unable to down migration version by 1 level: %v", err)
+		}
+		return nil
+	} else if opType == "upgrade_migration" {
+		if err := m.Steps(1); err != nil {
+			return fmt.Errorf("unable to down migration version by 1 level: %v", err)
```
[the remark this answers is not in the document]
Comment (tirthyme) [reply utilities/db-migration/migrate/migrate_db.go:56]: you're correct, it's a typo.

File: utilities/verify-db-migration.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+make db-migrate-downgrade
+make db-migrate-upgrade
+pg_dump postgresql://postgres:gitops@localhost:5432/postgres > "$MIGRATION_SQL_FILE" && cat "$MIGRATION_SQL_FILE"
+diff -Nur "$MIGRATION_SQL_FILE" "$MASTER_SQL_FILE"
+
+echo "Check if downgrading and upgrading by a two versions leads to same schema."
+make db-migrate-downgrade
+make db-migrate-downgrade
+make db-migrate-upgrade
+make db-migrate-upgrade
```
Comment (samyak-jn) [inline utilities/verify-db-migration.sh:53]: LGTM as well, thanks for catching the typo, Panos!
The only comment from my side, is this a redundant check and needed(?)

My quick suggestion would be to either keep one of them.

Comment (tirthyme) [reply utilities/verify-db-migration.sh:53]: I had discussed this with Jonathan before going on my vacation. Just to be on the safer side, we're checking if last 2 most recent migrations work properly after doing initial single migration check.

**Review by drpaneas — APPROVED** [review]
/lgtm thx

**Review by jgwest — APPROVED** [review]
Looks great, thanks @tirthyme!

**Pull request merged**